### DISCUSSION TITLE
How to match URLs with a colon in their name

### DISCUSSION THREAD

**Discussion by vlukashov** [pillarjs/path-to-regexp#274] [Q&A] [answered]
Colon (`:`) is a special character for `path-to-regexp`, so it's unclear how to write a route spec that allows using colons literally. For example, Wikipedia (and other media-wiki sites) uses colons in category page URLs: https://en.wikipedia.org/wiki/Category:JavaScript_libraries

**How would I write a route spec for such URLs with `path-to-regexp`?**

- My first assumption was that a colon can be escaped with a `\`, like in a regexp: `/wiki/Category\::category`. That did not work with `6.2.0`.
- My second assumption was that a colon can be repeated for its literal meaning, like `%` in URL-encoding: `/wiki/Category:::category`. That did not work with `6.2.0` either.

By now I assume that this class of URLs cannot be handled with `path-to-regexp`, and I'd be happy to learn that I'm wrong.

---
NOTE: A colon (`:`) might legitimately be used in the path part of a URL according to the [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3). It is actually used in the wild (Wikipedia).

"_Common_" characters that are NOT allowed in URL pathnames (according to my reading of the RFC) are `"`, `#`, `<`, `>`, `?`, `[`, `]`, `\`, `^`, `{`, `}`, ```​`​```. It would be great if `path-to-regexp` allowed to optionally change the character it uses in the path parameters syntax from the current `:` to one from the above list.

Am I overthinking this?

**Comment by dougwilson** [comment, marked as answer]
You can use backslash to escape the colon. The backslash needs to be in the string, ans since Javascript itself treats backslash as a special characher, to get a backslash in a string requires escaping it, like so: `'\\'`. This means a string of your path, witten as a Javascript literal, would need to be `'/wiki/Category\\::category'`

**Reply by vlukashov** [reply]
Indeed! Thanks for pointing out my mistake with a `\:` vs. `\\:`!
